appinsights/datacontracts.go

```diff
@@ -138,19 +156,29 @@ func (item *MetricTelemetry) baseData() Domain {
 	return item.data
 }
 
+func (item *MetricTelemetry) SetProperty(key, value string) {
+	if item.data.Properties == nil {
+		item.data.Properties = make(map[string]string)
+	}
+	item.data.Properties[key] = value
+}
+
 type RequestTelemetry struct {
 	BaseTelemetry
 	data *requestData
 }
 
-func NewRequestTelemetry(name string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
+func NewRequestTelemetry(name, httpMethod, url string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
 	now := time.Now()
```

Comment:
Data model changes are coming; and I think we'll still want to use RequestTelemetry since it will have a *typed* data contract vs Envelope which will necessarily be interface{} (object/void*).  [change from the future](https://github.com/jjjordanmsft/ApplicationInsights-Go/commit/396f92f9b96d2f20c1386b0f979e43c8c2e40771#diff-0cabe73fe8898c3b641b66ae1795caf4R107)